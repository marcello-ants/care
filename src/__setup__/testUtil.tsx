/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { NextRouter } from 'next/router';
import { render } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { RouterContext } from 'next/dist/shared/lib/router-context';
import DayJSUtils from '@date-io/dayjs';
import { theme } from '@care/material-ui-theme';
import { ThemeProvider } from '@material-ui/core';
import { FeatureFlags, FeatureFlagsProvider } from '@/components/FeatureFlagsContext';
import { AppStateProvider } from '@/components/AppState';
import { initialAppState } from '@/state';
import { MuiPickersUtilsProvider } from '@material-ui/pickers';
import { AppState } from '@/types/app';

export interface PrepareTreeOptions {
  pathname?: string;
  mocks?: MockedResponse[];
  flags?: FeatureFlags;
  appState?: AppState;
  addTypeName?: boolean;
  basePath?: string;
}

const prepareTree = (options?: PrepareTreeOptions) => {
  const {
    pathname = '/path',
    mocks = [],
    flags = {},
    appState = initialAppState,
    addTypeName = false,
    basePath = '/',
  } = options || {};
  const routerMock = {
    push: jest.fn(),
    asPath: pathname,
    beforePopState: (cb: any) => {
      cb({ url: pathname });
    },
    basePath,
    pathname,
    query: {},
    back: jest.fn(),
  } as unknown as NextRouter;

  const renderFn = (children: React.ReactNode) => (
    <ThemeProvider theme={theme}>
      <RouterContext.Provider value={routerMock}>
        <MockedProvider mocks={mocks} addTypename={addTypeName}>
          <MuiPickersUtilsProvider utils={DayJSUtils}>
            <FeatureFlagsProvider flags={flags}>
              <AppStateProvider initialStateOverride={appState}>{children}</AppStateProvider>
            </FeatureFlagsProvider>
          </MuiPickersUtilsProvider>
        </MockedProvider>
      </RouterContext.Provider>
    </ThemeProvider>
  );

  return {
    renderFn,
    routerMock,
  };
};

export const preRenderHook = (options?: PrepareTreeOptions) => {
  const tree = prepareTree(options);
  const renderFn = <T,>(hook: () => T) =>
    renderHook(hook, {
      wrapper({ children }: { children: React.ReactNode }) {
        return tree.renderFn(children);
      },
    });

  return {
    ...tree,
    renderFn,
  };
};

export const preRenderPage = (options?: PrepareTreeOptions) => {
  const tree = prepareTree(options);
  const renderFn = (ComponentOrChildren: React.ComponentType | React.ReactNode) => {
    if (React.isValidElement(ComponentOrChildren)) {
      return render(tree.renderFn(ComponentOrChildren));
    }

    const Component = ComponentOrChildren as React.ComponentType;
    return render(tree.renderFn(<Component />));
  };

  return {
    ...tree,
    renderFn,
    renderTree: tree.renderFn,
  };
};

export const setupWindowLocation = () => {
  const originalLocation = window.location;
  const windowLocationMock = jest.fn();

  // @ts-ignore
  delete window.location;
  // @ts-ignore
  window.location = {
    assign: windowLocationMock,
  };

  return {
    mock: windowLocationMock,
    cleanUp() {
      window.location = originalLocation;
    },
  };
};

export const setupLocalStorage = () => {
  const originalLocalStorage: Storage = window.localStorage;
  const localStorageMock = jest.fn();
  Object.defineProperty(window, 'localStorage', localStorageMock);

  return {
    mock: localStorageMock,
    cleanUp() {
      Object.defineProperty(window, 'localStorage', originalLocalStorage);
    },
  };
};

export function mockFetch({ ok, status, data }: { ok: boolean; status: number; data: any }) {
  return () =>
    Promise.resolve({
      ok,
      status,
      json: () => Promise.resolve(data),
    } as Response);
}
