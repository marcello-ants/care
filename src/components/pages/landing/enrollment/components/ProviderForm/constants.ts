export const PROVIDER_SERVICES_DESKTOP = [
  {
    description: 'Child care',
    label: 'Child care',
    value: 'CHILD_CARE',
  },
  {
    description: 'Senior care',
    label: 'Senior care',
    value: 'SENIOR_CARE',
  },
  {
    description: 'Housekeeping',
    label: 'Housekeeping',
    value: 'HOUSEKEEPING',
  },
  {
    description: 'Pet care',
    label: 'Pet care',
    value: 'PET_CARE',
  },
  {
    description: 'Tutoring',
    label: 'Tutoring',
    value: 'TUTORING',
  },
  {
    description: 'Special needs care',
    label: 'Special needs care',
    value: 'SPECIAL_NEEDS',
  },
  {
    description: 'Errands & odd jobs',
    label: 'Errands & odd jobs',
    value: 'GENERIC',
  },
];

export const PROVIDER_SERVICES_MOBILE = [
  {
    description: 'Child care',
    label: 'Child care',
    value: 'CHILD_CARE',
  },
  {
    description: 'Senior care',
    label: 'Senior care',
    value: 'SENIOR_CARE',
  },
  {
    description: 'Pet care',
    label: 'Pet care',
    value: 'PET_CARE',
  },
  {
    description: 'Tutoring',
    label: 'Tutoring',
    value: 'TUTORING',
  },
  {
    description: 'Special needs care',
    label: 'Special needs care',
    value: 'SPECIAL_NEEDS',
  },
  {
    description: 'Housekeeping',
    label: 'Housekeeping',
    value: 'HOUSEKEEPING',
  },
  {
    description: 'Errands & odd jobs',
    label: 'Errands & odd jobs',
    value: 'GENERIC',
  },
];

export const DROPOFF_URLS = {
  CHILD_CARE: '/swf/ccOptimizedFlow',
  SENIOR_CARE: '/app/enrollment/provider/sc ',
  PET_CARE: '/swf/pcOptimizedFlow',
  HOUSEKEEPING: '/swf/hkOptimizedFlow',
  TUTORING: '/swf/tuOptimizedFlow',
  SPECIAL_NEEDS: '/swf/snOptimizedFlow',
  GENERIC: '/swf/cgOptimizedFlow',
  AU_PAIR: '',
  CARE_GIGS: '',
};
