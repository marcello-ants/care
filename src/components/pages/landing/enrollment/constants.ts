export const SITTER_FORM = {
  providerServices: {
    title: 'What Services do you offer?',
    id: 'providerService',
    options: [
      {
        value: 'CHILD_CARE',
        text: 'Babysitter',
      },
      {
        value: 'CHILD_CARE',
        text: 'Nanny',
      },
      {
        value: 'SPECIAL_NEEDS',
        text: 'Special Needs Provider',
      },
      {
        value: 'TUTORING',
        text: 'Tutor',
      },
      {
        value: 'TUTORING',
        text: 'Private Lesson Instructor',
      },
      {
        value: 'SENIOR_CARE',
        text: 'Senior Care Provider',
      },
      {
        value: 'PET_CARE',
        text: 'Pet Care Provider',
      },
      {
        value: 'HOUSEKEEPING',
        text: 'Housekeeper',
      },
      {
        value: 'GENERIC',
        text: 'Errands &amp; Odd Jobs Provider',
      },
    ],
  },
  providerEmail: {
    id: 'email',
    placeholder: 'Email',
  },
  providerPassword: {
    id: 'password',
    placeholder: 'Password',
  },
  providerZip: {
    id: 'providerZip',
    placeholder: 'ZIP',
  },
};

export const GENERIC_FORM = {
  howDidYouHearAboutUs: {
    title: 'How did you hear about us?',
    id: 'howDidYouHearAboutUs',
    options: [
      {
        value: 'FACEBOOK',
        text: 'Facebook',
      },
      {
        value: 'BANNER_AD',
        text: 'Banner Ad',
      },
      {
        value: 'TV_AD',
        text: 'TV Ad',
      },
      {
        value: 'OTHER_SOCIAL_MEDIA',
        text: 'Other Social Media (Twitter, Instagram, Pinterest, LinkedIn)',
      },
      {
        value: 'PRESS_COVERAGE',
        text: 'Press Coverage (News, Magazine, Blog)',
      },
      {
        value: 'FRIENDS_OR_FAMILY',
        text: 'Friends or Family',
      },
      {
        value: 'ONLINE_VIDEO',
        text: 'Online Video (Hulu, Youtube)',
      },
      {
        value: 'SEARCH_ENGINE',
        text: 'Search Engine (Google, Bing)',
      },
      {
        value: 'PARENTING_GROUP',
        text: 'Parenting Group or Forum',
      },
      {
        value: 'OTHER',
        text: 'Other',
      },
    ],
  },
};
