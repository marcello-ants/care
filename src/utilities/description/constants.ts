export const EMPTY = '';
export const DOT = '.';
export const COMMA = ',';
export const SPACE = ' ';

export const CHILD_EXPECTING_ONLY = 'Looking for a caregiver for my newborn.';
export const INTRO = 'Looking for a caregiver for my';
export const EXPECTING = "I'm expecting!";
export const IDEA_CAREGIVER = 'Our ideal caregiver:';
export const WE_COULD_USE_HELP_WITH = 'We could use help with';

// Child care service
export const HOMEWORK_HELP = 'homework or curriculum assistance';
export const STRUCTURING_ACTIVITIES = 'structuring activities';

export const COMFORTABLE_WITH_PETS_LONG = 'Please be comfortable with pets.';
export const LIGHT_HOUSEKEEPING_LONG =
  'We would prefer someone who could help out with light housekeeping.';
export const OWN_CAR_LONG = 'It is important to us that you have your own car.';

export const SERVICES_INTRO = 'Additional needs include';
export const COMFORTABLE_WITH_PETS = 'being comfortable with pets';
export const LIGHT_HOUSEKEEPING = 'light housekeeping';
export const OWN_CAR = 'having a reliable car';
export const NON_SMOKER = 'a non-smoker';

// Child caregiver types
export const NANNY = 'nanny';
export const BABYSITTER = 'babysitter';

// Job schedule
export const FLEXIBLE_START_DATE = 'Flexible start date.';
export const VARYING_SCHEDUE = 'My schedule may vary.';
export const SCHEDULE_INTRO = 'Looking for care';

export const DAY_OF_WEEK_SORTER = {
  sunday: 0,
  monday: 1,
  tuesday: 2,
  wednesday: 3,
  thursday: 4,
  friday: 5,
  saturday: 6,
};
