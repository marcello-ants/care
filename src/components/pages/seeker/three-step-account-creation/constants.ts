import * as yup from 'yup';
import { ServiceType } from '@/__generated__/globalTypes';
import { VerticalsAbbreviation } from '@/constants';

export const validationSchema = yup.object({
  firstName: yup
    .string()
    .min(2, 'First name needs to have at least two letters.')
    .required('First name is required'),
  lastName: yup
    .string()
    .min(2, 'Last name needs to have at least two letters.')
    .required('Last name is required'),
});

export const SERVICE_TYPE_BY_VERTICAL: { [key: string]: ServiceType } = {
  HK: ServiceType.HOUSEKEEPING,
  PC: ServiceType.PET_CARE,
  TU: ServiceType.TUTORING,
  CC: ServiceType.CHILD_CARE,
  DC: ServiceType.CHILD_CARE,
  IB: ServiceType.CHILD_CARE,
  SC: ServiceType.SENIOR_CARE,
};

export const ERROR_CREATING_ACCOUNT = 'An error occurred creating your account, please try again.';

export const FACEBOOK_VERTICALS_ALLOWANCE = ['CC'];
export const HDYHAU_VERTICALS_ALLOWANCE: VerticalsAbbreviation[] = ['HK', 'TU', 'PC', 'DC'];
