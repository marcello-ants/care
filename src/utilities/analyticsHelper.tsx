/* istanbul ignore next */
import {
  Config,
  EventPayload,
  logEvent as _logEvent,
  init,
  setUserProps,
  getInstance,
} from '@care/analytics';
import dayjs from 'dayjs';
import getConfig from 'next/config';
import logger from '@/lib/clientLogger';
import { CLIENT_SIDE_ERROR_TAG } from '@/constants';
import { captureException, configureScope } from '@sentry/nextjs';
import { FeatureFlags } from '@/components/FeatureFlagsContext';
import { getCzenVisitorId, getCzenSessionId } from './czenCookiesHelper';
import getVertical from './verticalUtils';

const { publicRuntimeConfig } = getConfig();

export const AMPLITUDE_DATE_FORMAT = 'MMDDYYYY';
export const AMPLITUDE_TIME_FORMAT = 'hh:mm A';

let initialized = false;

/* istanbul ignore next */
function getCommonData() {
  const commonData: any = {
    job_flow: 'MW VHP enrollment',
    web_platform: window.innerWidth < 600 ? 'Mobile Web' : 'Desktop',
  };

  const czenSessionId = getCzenSessionId();
  if (czenSessionId) {
    commonData.czen_session_id = czenSessionId;
  }

  const czenVisitorId = getCzenVisitorId();
  if (czenVisitorId) {
    commonData.visitor_id = czenVisitorId;
  }

  return commonData;
}

const initialize = () => {
  if (!initialized) {
    const careAnalyticsConfig: Config = {
      analyticsEnabled: publicRuntimeConfig.ANALYTICS_ENABLED === 'true',
      key: publicRuntimeConfig.AMPLITUDE_API_KEY,
      commonData: getCommonData(),
    };
    init(careAnalyticsConfig);
    initialized = true;

    const amplitudeSessionId = getInstance()?.getSessionId();
    if (amplitudeSessionId) {
      configureScope((scope) => {
        const user = scope.getUser();
        scope.setUser({
          ...user,
          amplitudeSessionId,
        });
      });

      if (window.SplunkRum) {
        window.SplunkRum.setGlobalAttributes({
          'enduser.amplitudeSessionId': amplitudeSessionId,
        });
      }
    }
  }
};

interface AnalyticsHelperPayload extends EventPayload {
  vertical?: string;
}

export namespace AnalyticsHelper {
  /* istanbul ignore next */
  export const logEvent = (payload: AnalyticsHelperPayload) => {
    try {
      initialize();
      _logEvent({
        name: payload.name,
        data: { vertical: getVertical(), ...payload.data },
      });
    } catch (e: any) {
      logger.error({ error: e, tags: [CLIENT_SIDE_ERROR_TAG] });
      /* istanbul ignore next */
      captureException(e);
    }
  };

  /* istanbul ignore next */
  export const setDefaultUserProps = (careDeviceId: string) => {
    try {
      initialize();
      const userProps: any = {};
      const czenSessionId = getCzenSessionId();
      if (czenSessionId) {
        userProps.czen_session_id = czenSessionId;
      }
      const czenVisitorId = getCzenVisitorId();
      if (czenVisitorId) {
        userProps.visitor_id = czenVisitorId;
      }
      userProps.care_device_id = careDeviceId;
      setUserProps({
        props: userProps,
      });
    } catch (e: any) {
      logger.error({ error: e, tags: [CLIENT_SIDE_ERROR_TAG] });
      captureException(e);
    }
  };

  /* istanbul ignore next */
  export const setMemberId = (memberId: string) => {
    try {
      initialize();
      setUserProps({
        memberId: memberId || undefined,
      });
    } catch (e: any) {
      logger.error({ error: e, tags: [CLIENT_SIDE_ERROR_TAG] });
      captureException(e);
    }
  };

  /* istanbul ignore next */
  export const logTestExposure = (testName: string, evaluationDetail: FeatureFlags[string]) => {
    if (!evaluationDetail) return;

    // only log the exposure if the user matched the flag's inclusion rules,
    // in which case we also know that we have a non-null variationIndex since LD requires a variation
    // to be specified for each rule
    const loggableTestExposureReasons = ['RULE_MATCH', 'OVERRIDE'];
    if (loggableTestExposureReasons.includes(evaluationDetail.reason?.kind)) {
      AnalyticsHelper.logEvent({
        name: 'Test Exposure',
        data: {
          test_name: testName,
          test_variant: evaluationDetail.variationIndex!.toString(),
        },
      });
    }
  };

  /* istanbul ignore next */
  export const logScreenViewed = async (screenName: string, source: string) => {
    AnalyticsHelper.logEvent({
      name: 'Screen Viewed',
      data: {
        screen_name: screenName,
        source,
      },
    });
  };
}

export const formatAmplitudeDate = (date: string | null) => {
  if (date == null) return '';
  return dayjs(date).format(AMPLITUDE_DATE_FORMAT);
};

export const formatAmplitudeTime = (time: string) => {
  const [hour, minute] = time.split(':');
  return dayjs().hour(Number(hour)).minute(Number(minute)).format(AMPLITUDE_TIME_FORMAT);
};
