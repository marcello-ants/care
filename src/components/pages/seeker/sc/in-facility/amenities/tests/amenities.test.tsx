import { screen, fireEvent, waitFor } from '@testing-library/react';

import { SEEKER_IN_FACILITY_ROUTES } from '@/constants';
import { AnalyticsHelper } from '@/utilities/analyticsHelper';

import { SeniorCareFacilityAmenity } from '@/__generated__/globalTypes';
import { preRenderPage } from '@/__setup__/testUtil';

import AmenitiesPage from '../amenities';

describe('<Amenities /> Screen', () => {
  beforeAll(() => {
    jest.spyOn(AnalyticsHelper, 'logEvent');
  });

  it('Should match snapshot', () => {
    const { renderFn } = preRenderPage();
    const view = renderFn(AmenitiesPage);
    expect(view.asFragment()).toMatchSnapshot();
  });

  it('Should render correctly', () => {
    const { renderFn } = preRenderPage();
    const amenities = [
      'Pet friendly',
      'Art classes',
      'Exercise groups',
      'Community dining',
      'Library',
      'Beauty salon',
    ];
    renderFn(AmenitiesPage);

    expect(
      screen.getByText(
        /Communities often offer activities and amenities. Is your loved one interested in any of these?/
      )
    ).toBeInTheDocument();

    expect(screen.getByText(/Select all that apply./)).toBeInTheDocument();

    amenities.forEach((amenity) => {
      expect(screen.getByText(amenity)).toBeInTheDocument();
    });

    expect(screen.getByText('Next')).toBeInTheDocument();
  });

  it('Should call AnalyticsHelper.logEvent with correct data', async () => {
    const amenities = [
      SeniorCareFacilityAmenity.PET_FRIENDLY,
      SeniorCareFacilityAmenity.ART_CLASSES,
      SeniorCareFacilityAmenity.EXERCISE_GROUPS,
    ];
    const { renderFn } = preRenderPage();
    renderFn(AmenitiesPage);

    const petFriendly = screen.getByText(/Pet friendly/);
    fireEvent.click(petFriendly);
    const artClasses = screen.getByText(/Art classes/);
    fireEvent.click(artClasses);
    const exerciseGroups = screen.getByText(/Exercise groups/);
    fireEvent.click(exerciseGroups);

    const next = screen.getByText('Next');
    fireEvent.click(next);

    await waitFor(() =>
      expect(AnalyticsHelper.logEvent).toHaveBeenCalledWith({
        name: 'Member enrolled',
        data: {
          vertical: 'Seniorcare',
          enrollment_step: 'amenities',
          member_type: 'Seeker',
          cta_clicked: 'Next',
          amenities_selected: amenities.join(','),
        },
      })
    );
  });

  it('Should route to /recap screen', async () => {
    const { renderFn, routerMock } = preRenderPage();
    renderFn(AmenitiesPage);

    const next = screen.getByText('Next');
    fireEvent.click(next);

    expect(routerMock.push).toHaveBeenCalledWith(SEEKER_IN_FACILITY_ROUTES.RECAP);
  });
});
