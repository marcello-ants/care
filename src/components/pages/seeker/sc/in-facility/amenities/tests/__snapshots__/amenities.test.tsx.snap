// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Amenities /> Screen Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h2
        class="MuiTypography-root makeStyles-header-4 MuiTypography-h2"
      >
        Communities often offer activities and amenities. Is your loved one interested in any of these?
      </h2>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <p
        class="MuiTypography-root makeStyles-bodyText-2 MuiTypography-body2"
      >
        Select all that apply.
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <ul
        class="MuiList-root makeStyles-selector-1 MuiList-padding"
      >
        <li
          class="MuiListItem-root makeStyles-root-6 pill MuiListItem-gutters"
        >
          <input
            class="makeStyles-input-9"
            id="selector.undefined.0"
            name=""
            readonly=""
            type="checkbox"
            value="PET_FRIENDLY"
          />
          <div
            class="makeStyles-controlIndicator-10 md"
            tabindex="0"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap MuiGrid-align-items-xs-center"
            >
              <div>
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  Pet friendly
                </p>
              </div>
            </div>
            <div
              class="makeStyles-checkboxContainer-12"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-31 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-34"
                    data-indeterminate="false"
                    type="checkbox"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root makeStyles-root-35"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M5.6.75h12.8a4.8 4.8 0 014.85 4.85v12.8a4.8 4.8 0 01-4.85 4.85H5.6A4.8 4.8 0 01.75 18.4V5.6A4.93 4.93 0 015.6.75z"
                      fill="#fff"
                    />
                    <path
                      d="M18.4 23.27H5.6A4.88 4.88 0 01.73 18.4V5.6A4.93 4.93 0 015.6.73h12.8a4.88 4.88 0 014.87 4.87v12.8a4.88 4.88 0 01-4.87 4.87zM5.6 1.73A3.92 3.92 0 001.73 5.6v12.8a3.88 3.88 0 003.87 3.87h12.8a3.88 3.88 0 003.87-3.87V5.6a3.88 3.88 0 00-3.87-3.87z"
                      fill="#23405f"
                    />
                  </svg>
                </span>
              </span>
            </div>
          </div>
        </li>
        <li
          class="MuiListItem-root makeStyles-root-6 pill MuiListItem-gutters"
        >
          <input
            class="makeStyles-input-9"
            id="selector.undefined.1"
            name=""
            readonly=""
            type="checkbox"
            value="ART_CLASSES"
          />
          <div
            class="makeStyles-controlIndicator-10 md"
            tabindex="0"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap MuiGrid-align-items-xs-center"
            >
              <div>
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  Art classes
                </p>
              </div>
            </div>
            <div
              class="makeStyles-checkboxContainer-12"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-31 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-34"
                    data-indeterminate="false"
                    type="checkbox"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root makeStyles-root-36"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M5.6.75h12.8a4.8 4.8 0 014.85 4.85v12.8a4.8 4.8 0 01-4.85 4.85H5.6A4.8 4.8 0 01.75 18.4V5.6A4.93 4.93 0 015.6.75z"
                      fill="#fff"
                    />
                    <path
                      d="M18.4 23.27H5.6A4.88 4.88 0 01.73 18.4V5.6A4.93 4.93 0 015.6.73h12.8a4.88 4.88 0 014.87 4.87v12.8a4.88 4.88 0 01-4.87 4.87zM5.6 1.73A3.92 3.92 0 001.73 5.6v12.8a3.88 3.88 0 003.87 3.87h12.8a3.88 3.88 0 003.87-3.87V5.6a3.88 3.88 0 00-3.87-3.87z"
                      fill="#23405f"
                    />
                  </svg>
                </span>
              </span>
            </div>
          </div>
        </li>
        <li
          class="MuiListItem-root makeStyles-root-6 pill MuiListItem-gutters"
        >
          <input
            class="makeStyles-input-9"
            id="selector.undefined.2"
            name=""
            readonly=""
            type="checkbox"
            value="EXERCISE_GROUPS"
          />
          <div
            class="makeStyles-controlIndicator-10 md"
            tabindex="0"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap MuiGrid-align-items-xs-center"
            >
              <div>
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  Exercise groups
                </p>
              </div>
            </div>
            <div
              class="makeStyles-checkboxContainer-12"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-31 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-34"
                    data-indeterminate="false"
                    type="checkbox"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root makeStyles-root-37"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M5.6.75h12.8a4.8 4.8 0 014.85 4.85v12.8a4.8 4.8 0 01-4.85 4.85H5.6A4.8 4.8 0 01.75 18.4V5.6A4.93 4.93 0 015.6.75z"
                      fill="#fff"
                    />
                    <path
                      d="M18.4 23.27H5.6A4.88 4.88 0 01.73 18.4V5.6A4.93 4.93 0 015.6.73h12.8a4.88 4.88 0 014.87 4.87v12.8a4.88 4.88 0 01-4.87 4.87zM5.6 1.73A3.92 3.92 0 001.73 5.6v12.8a3.88 3.88 0 003.87 3.87h12.8a3.88 3.88 0 003.87-3.87V5.6a3.88 3.88 0 00-3.87-3.87z"
                      fill="#23405f"
                    />
                  </svg>
                </span>
              </span>
            </div>
          </div>
        </li>
        <li
          class="MuiListItem-root makeStyles-root-6 pill MuiListItem-gutters"
        >
          <input
            class="makeStyles-input-9"
            id="selector.undefined.3"
            name=""
            readonly=""
            type="checkbox"
            value="COMMUNITY_DINING"
          />
          <div
            class="makeStyles-controlIndicator-10 md"
            tabindex="0"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap MuiGrid-align-items-xs-center"
            >
              <div>
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  Community dining
                </p>
              </div>
            </div>
            <div
              class="makeStyles-checkboxContainer-12"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-31 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-34"
                    data-indeterminate="false"
                    type="checkbox"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root makeStyles-root-38"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M5.6.75h12.8a4.8 4.8 0 014.85 4.85v12.8a4.8 4.8 0 01-4.85 4.85H5.6A4.8 4.8 0 01.75 18.4V5.6A4.93 4.93 0 015.6.75z"
                      fill="#fff"
                    />
                    <path
                      d="M18.4 23.27H5.6A4.88 4.88 0 01.73 18.4V5.6A4.93 4.93 0 015.6.73h12.8a4.88 4.88 0 014.87 4.87v12.8a4.88 4.88 0 01-4.87 4.87zM5.6 1.73A3.92 3.92 0 001.73 5.6v12.8a3.88 3.88 0 003.87 3.87h12.8a3.88 3.88 0 003.87-3.87V5.6a3.88 3.88 0 00-3.87-3.87z"
                      fill="#23405f"
                    />
                  </svg>
                </span>
              </span>
            </div>
          </div>
        </li>
        <li
          class="MuiListItem-root makeStyles-root-6 pill MuiListItem-gutters"
        >
          <input
            class="makeStyles-input-9"
            id="selector.undefined.4"
            name=""
            readonly=""
            type="checkbox"
            value="LIBRARY"
          />
          <div
            class="makeStyles-controlIndicator-10 md"
            tabindex="0"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap MuiGrid-align-items-xs-center"
            >
              <div>
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  Library
                </p>
              </div>
            </div>
            <div
              class="makeStyles-checkboxContainer-12"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-31 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-34"
                    data-indeterminate="false"
                    type="checkbox"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root makeStyles-root-39"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M5.6.75h12.8a4.8 4.8 0 014.85 4.85v12.8a4.8 4.8 0 01-4.85 4.85H5.6A4.8 4.8 0 01.75 18.4V5.6A4.93 4.93 0 015.6.75z"
                      fill="#fff"
                    />
                    <path
                      d="M18.4 23.27H5.6A4.88 4.88 0 01.73 18.4V5.6A4.93 4.93 0 015.6.73h12.8a4.88 4.88 0 014.87 4.87v12.8a4.88 4.88 0 01-4.87 4.87zM5.6 1.73A3.92 3.92 0 001.73 5.6v12.8a3.88 3.88 0 003.87 3.87h12.8a3.88 3.88 0 003.87-3.87V5.6a3.88 3.88 0 00-3.87-3.87z"
                      fill="#23405f"
                    />
                  </svg>
                </span>
              </span>
            </div>
          </div>
        </li>
        <li
          class="MuiListItem-root makeStyles-root-6 pill MuiListItem-gutters"
        >
          <input
            class="makeStyles-input-9"
            id="selector.undefined.5"
            name=""
            readonly=""
            type="checkbox"
            value="BEAUTY_SALON"
          />
          <div
            class="makeStyles-controlIndicator-10 md"
            tabindex="0"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap MuiGrid-align-items-xs-center"
            >
              <div>
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  Beauty salon
                </p>
              </div>
            </div>
            <div
              class="makeStyles-checkboxContainer-12"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-31 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-34"
                    data-indeterminate="false"
                    type="checkbox"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root makeStyles-root-40"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M5.6.75h12.8a4.8 4.8 0 014.85 4.85v12.8a4.8 4.8 0 01-4.85 4.85H5.6A4.8 4.8 0 01.75 18.4V5.6A4.93 4.93 0 015.6.75z"
                      fill="#fff"
                    />
                    <path
                      d="M18.4 23.27H5.6A4.88 4.88 0 01.73 18.4V5.6A4.93 4.93 0 015.6.73h12.8a4.88 4.88 0 014.87 4.87v12.8a4.88 4.88 0 01-4.87 4.87zM5.6 1.73A3.92 3.92 0 001.73 5.6v12.8a3.88 3.88 0 003.87 3.87h12.8a3.88 3.88 0 003.87-3.87V5.6a3.88 3.88 0 00-3.87-3.87z"
                      fill="#23405f"
                    />
                  </svg>
                </span>
              </span>
            </div>
          </div>
        </li>
      </ul>
    </div>
    <div
      class="makeStyles-wrapper-41"
    >
      <div
        class="makeStyles-fade-42"
        style="position: relative; width: auto; margin-bottom: -24px; padding: 32px 40px 24px;"
      >
        <div
          class="makeStyles-buttonContainer-3"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-fullWidth"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Next
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
